ecl/eclcmd/eclcmd_core.cpp

```diff
@@ -1886,46 +1886,58 @@ class EclCmdZapGen : public EclCmdCommon
     }
     virtual int processCMD()
     {
-        Owned<IClientWsWorkunits> client = createCmdClient(WsWorkunits, *this);
-
-        Owned<IClientWUCreateZAPInfoRequest> zapReq = client->createWUCreateZAPInfoRequest();
-        zapReq->setWuid(optWuid.get());
-        zapReq->setIncludeThorSlaveLog(optIncThorSlave);
-        zapReq->setProblemDescription(optProblemDesc.get());
-
-        Owned<IClientWUCreateZAPInfoResponse> zapResp = client->WUCreateZAPInfo(zapReq);
-        int ret = outputMultiExceptionsEx(zapResp->getExceptions());
-        if (ret == 0)
+        if(optServer.isEmpty())
+            throw MakeStringException(-1, "Server IP not specified");
+        if(optWuid.isEmpty())
+            throw MakeStringException(-1, "WUID not specified");
+
+        //Create the file name for the output
+        StringBuffer outputFile;
+        if (!optPath.isEmpty())
         {
-            StringBuffer filePath;
-            if (optPath.length())
-            {
-                filePath.set(optPath);
-                const char *p = filePath.str();
-                p = (p + strlen(p) - 1);
-                if (!streq(p, PATHSEPSTR))
-                    filePath.append(PATHSEPSTR);
-            }
-            else
-                filePath.set(".").append(PATHSEPSTR);
-
-            filePath.append(zapResp->getZAPFileName());
-
-            const MemoryBuffer & zapFileData = zapResp->getThefile();
-
-            Owned<IFile> zapFile = createIFile(filePath.str());
-            Owned<IFileIO> zapFileIo = zapFile->open(IFOcreate);
-
-            size32_t written = zapFileIo->write(0, zapFileData.length(), (const void *)zapFileData.toByteArray());
-
-            if (written != zapFileData.length())
-                throw MakeStringException(-1, "truncated write to file %s, ZAP file creation cancelled.\n", filePath.str());
+            outputFile.set(optPath.get());
+            const char *p = outputFile.str();
+            p = (p + strlen(p) - 1);
+            if (!streq(p, PATHSEPSTR))
+                outputFile.append(PATHSEPSTR);
+        }
+        else
+            outputFile.set(".").append(PATHSEPSTR);
+        outputFile.append("ZAPReport_").append(optWuid.get());
+        if (!optUsername.isEmpty())
+            outputFile.append('_').append(optUsername.get());
+        outputFile.append(".zip");
+
+        //Create command URL
+        StringBuffer urlTail("/WUCreateAndDownloadZAPInfo?Wuid=");
+        urlTail.append(optWuid.get());
+        if (optIncThorSlave)
+            urlTail.append("&IncludeThorSlaveLog=true");
+        if (!optProblemDesc.isEmpty())
+            urlTail.append("&ProblemDescription=").append(optProblemDesc.get());
+        EclCmdURL eclCmdURL("WsWorkunits", !streq(optServer, ".") ? optServer : "localhost", optPort, optSSL, urlTail.str());
+
+        //Create CURL command
+        StringBuffer curlCommand = "curl -H \"Content-Type: application/octet-stream\"";
```

Comment:
I think this use of -H Content-Type: application/octet-stream

would be telling ESP that this request has that content type, not that you expect that content type in return.  Does the command work without that?

Reply:
Good point. That is not needed.